File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
		int product = 1;
		int count = 0;
		int latestZero = 0;
		for(int i = 0; i < nums.length; i++) {
			if(nums[i] != 0 ) product *= nums[i];
			else {count++; latestZero = i;}
		}
		if(count > 1) return new int[nums.length];
		int[] result = new int[nums.length];
		
		if(count == 1) {
			result[latestZero] = product;
		}
		
		if(count == 0) {
			for(int i = 0; i < nums.length; i++) {
				result[i] = product / nums[i];
			}
		}


		return result;
	}
}  

File: Data Structures & Algorithms/binary-search/submission-1.go
func search(nums []int, target int) int {
	result := -1
	l := 0
	r := len(nums) - 1

	for l <= r  {
		middle := (r-l) / 2 + l
		if nums[middle] == target {
			return middle
		} else if nums[middle] > target {
			r = middle - 1
		} else if nums[middle] < target {
			l = middle + 1
		}
	}

	
	return result
}
